R/BoostedStacking.R

```diff
@@ -0,0 +1,352 @@
+#' 
+#' @param model.multiplexer [\code{\link{ModelMultiplexer}}]\cr
+#'   The multiplexer learner.
+#' @param mm.ps Collection of parameters and their constraints for optimization.
+#'   Dependent parameters with a \code{requires} field must use \code{quote} and not
+#'   \code{expression} to define it.
+#' @param control [\code{\link{TuneControlRandom}}]\cr
+#'   Control object for search method. 
+#' @param resampling [\code{\link{ResampleDesc}} \cr
+#'   Resampling strategy.
+#' @param predict.type [\code{character(1)}]\cr 
+#'   Classification: \dQuote{response} (= labels) or \dQuote{prob} (= probabilities and labels by selecting the ones with maximal probability).
+#'   Regression: \dQuote{response} (= mean response) or \dQuote{se} (= standard errors and mean response).
+#'   par.set [\code{\link[ParamHelpers]{ParamSet}}]\cr
+#' @param measures [\code{\link{Measure}}]\cr
+#'   Performance measures.
+#' @param niter [\code{integer}]\cr
+#'   Number of boosting iterations.
+#' @param tolerance [\code{numeric}]\cr
+#'   Tolerance for stopping criterion.
+#' @param subsemble.prop  [\code{numeric}]\cr 
+#'   Proportion of data where tuning is run on. 
+#' @examples 
+#' \dontrun{
+#' lrns = list(
+#'   #makeLearner("classif.ksvm", kernel = "rbfdot"), # no implm for response and multiclass
+#'   makeLearner("classif.gbm"),
+#'   makeLearner("classif.randomForest"))
+#' mm = makeModelMultiplexer(lrns)
+#' ctrl = makeTuneControlRandom(maxit = 3L)
+#' ps = makeModelMultiplexerParamSet(mm,
+#'   makeIntegerParam("n.trees", lower = 1L, upper = 500L),
+#'   makeIntegerParam("interaction.depth", lower = 1L, upper = 10L),
+#'   makeIntegerParam("ntree", lower = 1L, upper = 500L),
+#'   makeIntegerParam("mtry", lower = 1L, upper = getTaskNFeats(pid.task)))
+#' lrns = lapply(lrns, setPredictType, "prob")
+#' stb = makeBoostedStackingLearner(model.multiplexer = mm,
+#'   predict.type = "prob", resampling = cv5, mm.ps = ps, control = ctrl,
+#'   measures = mmce, niter = 2L)
+#' r = resample(stb, task = pid.task, resampling = cv2)
+#' }
+#' @export  
+
+makeBoostedStackingLearner = function(model.multiplexer, mm.ps, 
+  control, resampling = cv2, predict.type = "prob",
+  measures, niter = 2L, tolerance = -0.1, subsemble.prop = 0.8) {
+  # checks
+  assertClass(model.multiplexer, "ModelMultiplexer")
+  assertClass(mm.ps, "ParamSet")
+  assertClass(control, "TuneControlRandom") # for now
+  assertClass(resampling, "ResampleDesc")
+  assertChoice(predict.type, choices = c("response", "prob"))
+  checkMeasures(measures)
+  assertInt(niter, lower = 1L)
+  assertNumber(tolerance)
+  # 
+  if (model.multiplexer$type == "classif" & model.multiplexer$predict.type == "response" & "factors" %nin% model.multiplexer$properties) {
+    stop("Base models in model multiplexer does not support classifcation with factor features, which are created by using predict.type='response' within base learners.")
+  }
+  # FIXME input check: measures and type
```

Comment:
please also check if you have FIXME's that are important (and maybe can be implemented fast), or need some discussion with bernd.